### PR TITLE
Initial soil model

### PR DESCRIPTION
**Pull request by simonbyrne**
Need to figure out the boundary conditions here.

### CODE REVIEW

File: examples/Land/soil.jl [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+# Set up DG scheme
+dg = DGModel( # 
+  m, # "PDE part"
+  grid,
+  CentralNumericalFluxNonDiffusive(), # penalty terms for discretizations
+  CentralNumericalFluxDiffusive(),
+  CentralGradPenalty())
+
+
+Δ = min_node_distance(grid)
```
Comment (tapios) [inline examples/Land/soil.jl]: Do you want to specify that what matters here is the vertical direction?

File: examples/Land/soilmodel.jl [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+function gradvariables!(m::SoilModel, transform::Vars, state::Vars, aux::Vars, t::Real)
+  transform.T = state.ρcT / (m.ρ * m.c)
+end
+function diffusive!(m::SoilModel, diffusive::Vars, ∇transform::Grad, state::Vars, aux::Vars, t::Real)
+  diffusive.∂T∂z = ∇transform.T[2]
+end
+function flux_nondiffusive!(m::SoilModel, flux::Grad, state::Vars, aux::Vars, t::Real)
+end
+function flux_diffusive!(m::SoilModel, flux::Grad, state::Vars, diffusive::Vars, aux::Vars, t::Real)
+  flux.ρcT += SVector(0, m.λ * diffusive.∂T∂z, 0)
```
Comment (tapios) [inline examples/Land/soilmodel.jl]: Maybe better to make this coordinate independent, using the `vertical_unit_vector` from `orientation.jl`, i.e., 
```suggestion
  flux.ρcT += m.λ * diffusive.∂T∂z .* vertical_unit_vector
```
(To make sure this works on the sphere in the end.)

**Comment by tapios** [comment]
Looks good. You can add the boundary conditions when the more general B.C. Interface is sorted out.

File: examples/Land/soilmodel.jl [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                        init_aux!, init_state!,
+                        boundary_state!, wavespeed, LocalGeometry
+using CLIMA.DGmethods.NumericalFluxes: NumericalFluxNonDiffusive,
+                                       NumericalFluxDiffusive,
+                                       GradNumericalPenalty
+
+"""
+    SoilModel
+
+Computes diffusive flux `F` in:
```
Comment (tapios) [inline examples/Land/soilmodel.jl]:
```suggestion
Computes diffusive flux on the right-hand side of the diffusion equation:
```

**Comment by tapios** [comment]
Can we merge this so others can take over?

**Comment by charleskawczynski** [comment]
@gagnelandmanna and I are working on this

**Comment by charleskawczynski** [comment]
@kmdeck @gagnelandmanna, this has gotten quite stale. Can we close this PR? Or are there things in here we should keep?

**Comment by kmdeck** [comment]
> @kmdeck @gagnelandmanna, this has gotten quite stale. Can we close this PR? Or are there things in here we should keep?
@charleskawczynski 
IMO we can close. It looks like Elias wrote his own finite difference algorithm for soil, which might be useful to hang onto: https://github.com/CliMA/ClimateMachine.jl/blob/852386c495c0261ab40a7b81a408dfa8cb016670/examples/Land/Soil_Physics.jl#L95

Some references mentioned that could be useful to know of:
Dai et al (2019a) and Balland and Arp (2005) 

Other capabilities we have in the new framework.

We can wait for @gagnelandmanna to confirm though.

Does closing  = deleting? Should i grab a copy of that file I mentioned?

**Comment by charleskawczynski** [comment]
> Does closing = deleting? Should i grab a copy of that file I mentioned?

I don't think so, but it would be good to create a local branch so that someone doesn't delete the local branch when house-cleaning.

**Comment by kmdeck** [comment]
> > Does closing = deleting? Should i grab a copy of that file I mentioned?
> 
> I don't think so, but it would be good to create a local branch so that someone doesn't delete the local branch when house-cleaning.

OK, I made  a local copy.

**Pull request closed without merging**